CONTRIBUTING.md

```diff
@@ -132,8 +131,39 @@ If it has someone assigned to it then that person is the person responsible for
 
-If you find an issue with an existing translation, please see the [Reporting Issues](#reporting-issues) section to open an issue and our translation team will work with you on a resolution. 
+If you find an issue with an existing translation, please see the [Reporting Issues](#reporting-issues) section to open an issue and our translation team will work with you on a resolution.
+
+### Common Git workflows
+
+In general there are two separate git workflows to follow when contributing code for a net new feature to AWX. One for keeping a feature branch up to date with devel, and another, used by individual contributors to keep their working branch up to date with the feature branch they are working on. When initiating work on a net new feature a new feature branch should be checked out from awx/devel and branch protection should be activated. Checking out the new branch, and activating branch protection must be done by someone with admin privileges on the AWX repo. This branch will be used by the developers working on the feature and merging work into the feature branch will allow the developers to work quickly without impacting devel. This also means that those working on the feature can decide what requirements need to pass before merging work into the feature branch. Merge meeting, QE sign off etc. can all be done at 1 time when the feature is ready to merge into awx/devel.
+
+It is important that the feature branch be kept up-to-date with awx/devel to limit merge conflicts that might arise when merging the feature into devel. To do this someone with admin privileges on the AWX repo needs to remove branch protection and then follow these steps (In this example I will use origin as the remote name for AWX):
+
+- `git fetch origin`
+- `git merge origin/devel --ff-only` merges the branch you are on with awx/devel and fast forwards.
+
+If there are merge conflicts we recommend not using `git merge origin/devel -ff-only` and instead rebasing. Those steps are:
+
+- `git fetch origin` (you might have done this step above so it is not necessary to repeate it here)
+- `git rebase -i origin/devel` (this will do an interactive rebase and make it easier to work through any merge conflicts)
 
+Then you are done and your feature branch is up-to-date with awx/devel. Just push your feature branch to `awx/feature-branch`.
+The second workflow should be used by individual contributors to keep their working branch up-to-date with the feature branch. Those steps are:
+
+- `git fetch upstream`
+- `git checkout origin/feature-branch`
```

Comment:
I assume that the inclusion of the remote here is to avoid the situation where the local checkout of the feature branch becomes out of date.  If so, the remote ought to match the git fetch command above (I'd recommend going with `upstream` for both, per my earlier comment).  If not, just go with `git checkout feature-branch`, but you'll need to include a command to bring it up to date with the official version (just `git merge upstream/feature-branch` is what I do.  Include `--ff-only` if paranoid.)